### PR TITLE
remove docker-generate and docker-notation code

### PR DESCRIPTION
**Pull request by chloeyin**
Signed-off-by: zaihaoyin <zaihaoyin@microsoft.com>

This PR removes `docker-notation` and `docker-generate` command as discussed in 
- Closes https://github.com/notaryproject/notation/issues/254. 
It also does some clean up jobs.

### CODE REVIEW

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: internal/cmd/flags.go [hunk 1]
```diff
@@ -8,51 +8,6 @@ import (
 	"time"
 
 	"github.com/spf13/pflag"
-	"github.com/urfave/cli/v2"
```
Comment (dtzar) [inline internal/cmd/flags.go:11]: Why do we still use urfave/cli?

Comment (shizhMSFT) [reply internal/cmd/flags.go:11]: That’s why we need this PR to clean up dead package code.

**Review by SteveLasker — APPROVED** [review]
a step forward

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by dtzar — APPROVED** [review]

**Pull request merged**